core/core-impl/src/main/java/ai/timefold/solver/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -90,9 +91,10 @@ public class EntityDescriptor<Solution_> {
     private final SolutionDescriptor<Solution_> solutionDescriptor;
     private final Class<?> entityClass;
     private final Predicate<Object> isInitializedPredicate;
-    private final Predicate<Object> hasNoNullVariables;
     private final List<MemberAccessor> declaredPlanningPinIndexMemberAccessorList = new ArrayList<>();
 
+    private Predicate<Object> hasNoNullVariablesBasicVar;
+    private Predicate<Object> hasNoNullVariablesListVar;
```

Comment:
Good question. In this case, I decided to go for `null`, because that's what those predicates test for. The meaning to that null check is assigned later; at this level, null is all there is.